python/anvill/binja.py

```diff
@@ -238,16 +238,52 @@ def _convert_bn_type(tinfo: bn.types.Type, cache):
         width = tinfo.width
         return FloatingPointType(width)
 
+    elif tinfo.type_class == bn.TypeClass.NamedTypeReferenceClass:
+        named_tinfo = tinfo.named_type_reference
+        if (named_tinfo.type_class
+            == bn.NamedTypeReferenceClass.StructNamedTypeClass):
+            # Add array of size width as struct element type
+            # TODO(akshayk) How to recover the struct elements type??
+            ret = StructureType()
+            elem_type_ = ArrayType()
+            elem_type_.set_num_elements(tinfo.width)
+            ret.add_element_type(elem_type_)
+            cache[str(named_tinfo)] = ret
+            return ret
+
+        elif (named_tinfo.type_class
+            == bn.NamedTypeReferenceClass.UnionNamedTypeClass):
+            # recover union type as block of bytes type
+            ret = UnionType()
+            elem_type_ = ArrayType()
+            elem_type_.set_num_elements(tinfo.width)
+            ret.add_element_type(elem_type_)
+            cache[str(tinfo)] = ret
+            return ret
+
+        elif (named_tinfo.type_class
+            == bn.NamedTypeReferenceClass.TypedefNamedTypeClass):
+            ret = TypedefType()
```

Comment:
I feel like you need to more here? Like, fill in the typedef name, or what it refers to?

Reply:
Probably using `ret.set_underlying_type`.

Reply:
I agree there is a need to do more with Typedef and Enum Named Types. I was not getting too many references about getting alias type of Named type references. Need to look more.